runtimes/runtimes/chat/encryptedChat.ts

```diff
@@ -0,0 +1,156 @@
+import { jwtDecrypt, EncryptJWT, JWTPayload } from 'jose'
+import { Connection } from 'vscode-languageserver'
+import {
+    ChatParams,
+    ChatResult,
+    EndChatParams,
+    FeedbackParams,
+    FollowUpClickParams,
+    InfoLinkClickParams,
+    InsertToCursorPositionParams,
+    LinkClickParams,
+    NotificationHandler,
+    QuickActionParams,
+    QuickActionResult,
+    RequestHandler,
+    SourceLinkClickParams,
+    TabAddParams,
+    TabChangeParams,
+    TabRemoveParams,
+    chatRequestType,
+    feedbackNotificationType,
+    readyNotificationType,
+    tabAddNotificationType,
+    tabChangeNotificationType,
+    tabRemoveNotificationType,
+    insertToCursorPositionNotificationType,
+    linkClickNotificationType,
+    infoLinkClickNotificationType,
+    sourceLinkClickNotificationType,
+    followUpClickNotificationType,
+    endChatRequestType,
+    CancellationToken,
+    EncryptedChatParams,
+    EncryptedQuickActionParams,
+    quickActionRequestType,
+} from '../../protocol'
+import { Chat } from '../../server-interface'
+import { CredentialsEncoding } from '../auth/standalone/encryption'
+
+export class EncryptedChat implements Chat {
+    private key: Buffer
+    private encoding: CredentialsEncoding | undefined
+
+    constructor(
+        private readonly connection: Connection,
+        key: string,
+        encoding?: CredentialsEncoding
+    ) {
+        this.key = Buffer.from(key, 'base64')
+        this.encoding = encoding
+    }
+
+    public onChatPrompt(handler: RequestHandler<ChatParams, ChatResult | null | undefined, ChatResult>) {
+        this.connection.onRequest(chatRequestType, async (request: ChatParams | EncryptedChatParams) => {
```

Comment:
Does this have to be `ChatParams | EncryptedChatParams`? At this point we're only dealing in `EncryptedChatParams`, right?

Reply:
I know that this `chatRequestType` defines the union type, but could we simply not do that here but define only an `encryptedChatRequestType`?

Reply:
We are only dealing with EncryptedParams but the signature for `chatRequestType` expect both of these types, otherwise typescript complains